### PR TITLE
feat: load more updates

### PR DESCRIPTION
**Pull request by roger-in-kiva**
- load more journal updates feature

Screenshot 2024-11-21 at 1 51 01 p m

### CODE REVIEW

File: src/pages/Portfolio/MyKiva/MyKivaPage.vue [hunk 1]
```diff
@@ -255,18 +257,25 @@ const handleBackToJourney = () => {
...
 			totalUpdates.value = result.data?.lend?.loan?.updates?.totalCount ?? 0;
+			const updates = result.data?.lend?.loan?.updates?.values ?? [];
+			loanUpdates.value = loanUpdates.value.concat(updates);
 		}).catch(e => {
 			logReadQueryError(e, 'MyKivaPage updatesQuery');
 		});
 };
 
+const loadMoreUpdates = () => {
+	updatesOffset.value += 1;
```
Comment (dyersituations) [inline src/pages/Portfolio/MyKiva/MyKivaPage.vue:271]: I guess this is what we're chatting about in slack. This probably needs to be 3, right?

File: src/components/MyKiva/JournalUpdatesCarousel.vue [hunk 2]
```diff
@@ -19,6 +20,20 @@
...
+				<div
+					:key="`view-more-card`"
+					class="tw-flex tw-items-center tw-h-full tw-pr-3"
+				>
+					<kv-button
+						class="tw-mt-2"
+						variant="secondary"
+						@click="loadMoreUpdates"
+					>
+						Load more<br>updates
```
Comment (dyersituations) [inline src/components/MyKiva/JournalUpdatesCarousel.vue:33]: Does this `br` also look right in mobile?

File: src/components/MyKiva/JournalUpdatesCarousel.vue [hunk 3]
```diff
@@ -118,11 +139,24 @@ const interactCarousel = () => {
...
+
+watch(
+	() => updates,
+	() => {
+		if (updates.value.length > 0 && updates.value.length < 3) {
+			$kvTrackEvent('portfolio', 'view', 'At least one journal update viewed');
+			carouselIndex.value = 0;
+		}
+		if (updates.value.length > 3) {
+			carouselIndex.value = updates.value.length - 2;
```
Comment (dyersituations) [inline src/components/MyKiva/JournalUpdatesCarousel.vue:155]: What does the `- 2` here do exactly? Focus the newly loaded last update?

File: src/components/MyKiva/JournalUpdatesCarousel.vue (hunk 2 above)
Comment (roger-in-kiva) [reply src/components/MyKiva/JournalUpdatesCarousel.vue:33]: Yeah, makes a smaller button there.

**Review by dyersituations — APPROVED** [review]

**Pull request merged**